src/main/java/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -225,7 +225,7 @@ void undoBattle(final Route route, final Collection<Unit> units, final PlayerID
     // We must look at all territories,
     // because we could have conquered the end territory if there are no units there
     for (final Territory current : route.getAllTerritories()) {
-      if (!relationshipTracker.isAllied(current.getOwner(), player) && m_conquered.contains(current)) {
+      if (!relationshipTracker.isAllied(player, current.getOwner()) && m_conquered.contains(current)) {
```

Comment:
Strange...  It appears 84f7868 is empty and didn't actually revert this line.  Maybe there was a rebase in between?  Regardless, could you please push a new commit to revert this line?